service/controller/v13/resource/node/create.go

```diff
@@ -0,0 +1,125 @@
+package node
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/client/k8srestconfig"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/api/v1/node"
+
+	"github.com/giantswarm/kvm-operator/service/controller/v13/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// At first we need to create a Kubernetes client for the reconciled guest
+	// cluster.
+	var k8sClient kubernetes.Interface
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "creating K8s client for the guest cluster")
+
+		certs, err := r.certsSearcher.SearchCluster(key.ClusterID(customObject))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c := k8srestconfig.Config{
+			Logger: r.logger,
+
+			Address:   key.ClusterAPIEndpoint(customObject),
+			InCluster: false,
+			TLS: k8srestconfig.TLSClientConfig{
+				CAData:  certs.APIServer.CA,
+				CrtData: certs.APIServer.Crt,
+				KeyData: certs.APIServer.Key,
+			},
+		}
+		restConfig, err := k8srestconfig.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		k8sClient, err = kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "created K8s client for the guest cluster")
+	}
+
+	// We need to fetch the nodes being registered within the guest cluster's
+	// Kubernetes API. The list of nodes is used below to sort out which ones have
+	// to be deleted if there does no associated host cluster pod exist.
+	var nodes []corev1.Node
+	{
+		list, err := k8sClient.CoreV1().Nodes().List(metav1.ListOptions{})
+		if IsGuestAPINotAvailable(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "guest cluster is not available")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
```

Comment:
Is resource cancelling missing here or should the above log message be changed_

Reply:
Canceling works with the default resource by simply returning. The context cancelation has to be used with CRUD resources. This here is no CRUD resource. See also https://github.com/giantswarm/operatorkit/blob/master/docs/control_flow_primitives.md#default-resources.